ReSharper.FSharp/src/FSharp.Psi.Intentions/src/Intentions/UseNamedAccessAction.fs

```diff
@@ -0,0 +1,103 @@
+open type JetBrains.Diagnostics.Assertion
+
+[<ContextAction(Name = "UseNamedAccess", Group = "F#", Description = "Use named access inside a DU pattern")>]
+type UseNamedAccessAction(dataProvider: FSharpContextActionDataProvider) =
+    inherit FSharpContextActionBase(dataProvider)
+    
+    let mutable names = Array.empty
+    let mutable tuplePat : ITuplePat = null
+
+    override x.Text = "Use named access"
+
+    override x.IsAvailable _ =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>()
+        if isNull pattern then false else
+        // We expect a single IParenPat here
+        if pattern.Parameters.Count <> 1 then false else
+        let parenPat =
+            match pattern.Parameters.[0] with
+            | :? IParenPat as parenPat -> parenPat
+            | _ -> null
+
+        if isNull parenPat then false else
+
+        // Ignore multiline patterns for now
+        if parenPat.StartLine <> parenPat.EndLine then false else
+
+        // This should only work when there are multiple items
+        tuplePat <- parenPat.Pattern :?> ITuplePat
+        if isNull tuplePat then false else
+
+        // We need to be sure that the pattern is a DU with all named fields
+        let fcsUnionCase = pattern.ReferenceName.Reference.GetFcsSymbol().As<FSharpUnionCase>()
+        if isNull fcsUnionCase then false else
+            
+        names <-
+            fcsUnionCase.Fields
+            |> Seq.choose (fun field -> if field.IsNameGenerated then None else Some field.Name)
+            |> Seq.toArray
+        
+        // All fields need to be named
+        if names.Length <> fcsUnionCase.Fields.Count then false else
+        // The amount of fields needs to match with the tuple length
+        if names.Length <> tuplePat.Patterns.Count then false else
+
+        true
+
+    override x.ExecutePsiTransaction _ : unit =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>().NotNull()
+        use writeCookie = WriteLockCookie.Create(pattern.IsPhysical())
+        use disableFormatter = new DisableCodeFormatter()
+        
+        let rec visit (pat: IFSharpPattern) =
+            match pat.IgnoreInnerParens() with
+            | :? IWildPat-> None
+            | :? ITuplePat as tuplePat ->
+                let hasAnyPatterns = tuplePat.Patterns |> Seq.exists (visit >> Option.isSome)
+                if hasAnyPatterns then Some pat else None
+            | _ -> Some pat
+        
+        let usedFieldsWithPatterns: (string * IFSharpPattern)[] =
+            (names, tuplePat.PatternsEnumerable)
+            ||> Seq.zip
+            |> Seq.choose (fun (name, pat) -> visit pat |> Option.map (fun pat -> name, pat))
+            |> Seq.toArray
+
+        let factory = pattern.CreateElementFactory()
+        let sourceText =
+            let fields =
+                usedFieldsWithPatterns
+                |> Array.map (fun (name,_) -> $"{name} = _")
```

Comment:
Is there a Fantomas setting that controls the spaces around `=`?

Reply:
No, the style guide mentions spaces explicitly ([source](https://learn.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-named-arguments))